url82787aa7a4_vrthra_v_language_tgz-pJ8-v_VJ8/src/v/Token.java

```
@@ -0,0 +1,81 @@
+    }
+    public char cvalue() {
+        try {
+            Term<Character> v = (Term<Character>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:bool",this,"Invalid type(need bool)");
+        }
+    }
+    public int ivalue() {
+        try {
+            Term<Integer> v = (Term<Integer>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:int",this,"Invalid type(need int)");
+        }
+    }
+    public double dvalue() {
+        try {
+            Term<Double> v = (Term<Double>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:decimal",this,"Invalid type(need decimal)");
+        }
+    }
+    public Number numvalue() {
+        if (type() == Type.TInt)
+            return new Integer(ivalue());
+        if (type() == Type.TDouble)
+            return new Double(dvalue());
+        throw new VException("err:type:number",this,"Invalid type(NaN)");
+    }
+    public String svalue() {
+        try {
+            Term<String> v = (Term<String>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:string",this,"Invalid type(need string)");
+        }
+    }
+    public Quote qvalue() {
+        try {
+            Term<Quote> v = (Term<Quote>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:quote",this,"Invalid type(need quote)");
+        }
+    }
+    public Object ovalue() {
+        try {
+            Term<Object> v = (Term<Object>)this;
+            return v.val;
+        } catch (Exception e) {
+            throw new VException("err:type:object",this,"Invalid type(need object)");
+        }
+    }
+// meta
+    public VFrame fvalue() {
+        try {
+            Term<VFrame> v = (Term<VFrame>)this;
+            return v.val;
+        } catch (Exception e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)